Ensure MacPorts libiconv and icu are found even if -I/opt/local/include
is not in CPPFLAGS. Also don't expose the ICU C++ API if using a C++
version that is too old.
--- include/libxml/encoding.h.orig	2017-11-14 08:00:17.000000000 +1100
+++ include/libxml/encoding.h	2019-11-21 19:10:55.000000000 +1100
@@ -25,10 +25,14 @@
 #include <libxml/xmlversion.h>
 
 #ifdef LIBXML_ICONV_ENABLED
-#include <iconv.h>
+#include <@ICONV_PREFIX@/include/iconv.h>
 #endif
 #ifdef LIBXML_ICU_ENABLED
-#include <unicode/ucnv.h>
+/* ICU C++ API requires C++11 */
+#if defined(__cplusplus) && __cplusplus < @ICU_CXX_VERSION@
+#define U_SHOW_CPLUSPLUS_API 0
+#endif
+#include <@ICU_PREFIX@/include/unicode/ucnv.h>
 #endif
 #ifdef __cplusplus
 extern "C" {
